test/v2/core/wsgETH.spec.js

```diff
@@ -0,0 +1,198 @@
+  });
+
+  it("price per share", async () => {
+    const splitterAddresses = [deployer.address, multiSig.address, wsgEth.address];
+    const splitterValues = [6, 3, 31];
+
+    const PaymentSplitter = await ethers.getContractFactory("PaymentSplitter");
+    const paymentSplitter = await PaymentSplitter.deploy(splitterAddresses, splitterValues);
+    await paymentSplitter.deployed();
+
+    const rolloverVirtual = "1080000000000000000";
+    const vETH2Addr = "0x898bad2774eb97cf6b94605677f43b41871410b1";
+
+    const Withdrawals = await ethers.getContractFactory("Withdrawals");
+    const withdrawals = await Withdrawals.deploy(vETH2Addr, rolloverVirtual);
+    await withdrawals.deployed();
+
+    const numValidators = 1000;
+    const adminFee = 0;
+
+    const addresses = [
+      ethers.constants.AddressZero, // fee splitter
+      sgEth.address, // sgETH address
+      wsgEth.address, // wsgETH address
+      multiSig.address, // government address
+      ethers.constants.AddressZero, // deposit contract address - can't find deposit contract - using dummy address
+    ];
+
+    // add secondary minter contract / eoa
+    const Minter = await ethers.getContractFactory("SharedDepositMinterV2");
+    const minter = await Minter.deploy(numValidators, adminFee, addresses);
+    await minter.deployed();
+
+    const RewardsReceiver = await ethers.getContractFactory("RewardsReceiver");
+    const rewardsReceiver = await RewardsReceiver.deploy(withdrawals.address, [
+      sgEth.address,
+      wsgEth.address,
+      paymentSplitter.address,
+      minter.address,
+    ]);
+    await rewardsReceiver.deployed();
+
+    await sgEth.addMinter(minter.address);
+
+    // approve sgEth to alice
+    await sgEth.approve(wsgEth.address, parseEther("2"));
+    await wsgEth.deposit(parseEther("2"), alice.address);
+
+    console.log(formatEther(await wsgEth.pricePerShare()));
+
+    await expect(wsgEth.connect(alice).redeem(parseEther("0.5"), alice.address, alice.address))
+      .to.be.emit(wsgEth, "Withdraw")
+      .withArgs(alice.address, alice.address, alice.address, parseEther("0.5"), parseEther("0.5"));
+
+    // deposit to rewardReceiver to simulate reward
+    await deployer.sendTransaction({
+      to: rewardsReceiver.address,
+      value: parseEther("1"),
+    });
+    // sends 60% of sgEth to WSGEth contract - so current rate is 1.5/2.1
+    console.log(await rewardsReceiver.state());
+    await rewardsReceiver.work();
+
+    await expect(wsgEth.syncRewards()).to.be.revertedWith("SyncError()");
+    // increase time by reward cycle
+    await time.increase(24 * 60 * 60);
+    await wsgEth.syncRewards();
+
+    await time.increase(24 * 60 * 60);
+
+    // redeem will get 1.1 sgEth
+    await expect(wsgEth.connect(alice).redeem(parseEther("0.5"), alice.address, alice.address))
+      .to.be.emit(wsgEth, "Withdraw")
+      .withArgs(alice.address, alice.address, alice.address, parseEther("0.7"), parseEther("0.5"));
```

Comment:
are we checking the invariant? i.e. alice has the expected balance after withdraw? 
im not sure how the 10% bonus is represented here, 0.5 -> 0.55 would be expected based on the comment

also are you sure you need to manually call syncRewards? 
iirc it should be triggered on redeem.

Reply:
with my analysis, reward of n' epoch distributes to next round by linearly. To test easily, I should call this once.